connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/InboundIntermediateConnectorContextImpl.java

```diff
@@ -91,6 +91,11 @@ public CorrelationResult correlateWithResult(Object variables) {
     return inboundContext.correlateWithResult(variables);
   }
 
+  @Override
+  public CorrelationResult correlateWithRequest(CorrelationRequest correlationRequest) {
+    return inboundContext.correlateWithResult(correlationRequest.getVariables());
```

Comment:
this one should use `correlate` internally.

```suggestion
    return inboundContext.correlate(correlationRequest);
```